MultiMarginCriterion.lua

```diff
@@ -9,10 +9,36 @@ end
 
 function MultiMarginCriterion:updateOutput(input, target)
    -- backward compatibility
+   local _target = target
+   if not torch.isTensor(_target) then
+     _target = input.new(1)
```

Comment:
if `target` is a number, this allocates memory at each iteration (although it's only a 1-element tensor...)